openmc/data/neutron.py

```diff
@@ -480,6 +480,12 @@ def export_to_hdf5(self, path, mode='a', libver='earliest'):
                 tgroup = g.create_group('total_nu')
                 rx.derived_products[0].to_hdf5(tgroup)
 
+        # Write redundant reaction data only for reactions with photon production
+        for rx in self.redundant_reactions.values():
+            if any([p.particle == 'photon' for p in rx.products]):
```

Comment:
Passing a generator expression here would avoid the creation of an extra list:
```Python
if any(p.particle == 'photon' for p in rx.products):
    ...
```
Note that when a generator expression is the only argument in a function call, it's not necessary to put an extra set of parentheses around it